os/kernel/binary_manager/binary_manager_verify.c

```diff
@@ -180,7 +180,8 @@ int binary_manager_read_header(int type, char *devpath, void *header_data, bool
 		while (bin_size > 0) {
 			read_size = bin_size < crc_bufsize ? bin_size : crc_bufsize;
 			ret = read(fd, (void *)crc_buffer, read_size);
-			if (ret != read_size) {
+			/* prevent going into infinite loop if readsize is zero */
+			if (ret != read_size && read_size == 0) {
```

Comment:
I don't know this is necessary.
1. by line 181, read_size could be the bin_size, but it's checked in line 180. It can't be zero
2. by line 181, read_size could be the crc_bufsize, but if crc_bufsize is zero, then by kmm_malloc, line 172, it will be returned at line 173. (malloc(0) returns null)